graylog2-web-interface/src/components/visualizations/QuickValuesVisualization.jsx

```diff
@@ -27,15 +27,25 @@ const QuickValuesVisualization = React.createClass({
     horizontal: PropTypes.bool,
     displayAnalysisInformation: PropTypes.bool,
     displayAddToSearchButton: PropTypes.bool,
+    sortOrder: PropTypes.string,
+    dataTableTitle: PropTypes.string,
+    limit: PropTypes.number,
+  },
+  getDefaultProps() {
+    return {
+      dataTableTitle: 'Top values',
+      limit: 50,
```

Comment:
This setting is being currently misused as far as I can see. We use it to set the table size, which I guess it's what we want. At the same time, we also use it to calculate the top values and as a cap for the number of slices in the pie chart.

Reply:
By the way, the changes removed all usages of the `NUMBER_OF_TOP_VALUES` variable, but the variable itself is still in the code. We should remove it if we don't need it any longer.

Reply:
Actually all three are required. The idea is to provide a Top-N widget, where n is the limit of the number of top items. I do not know if the `.size(limit)` statement is required, because the top group should also include limiting the table size, obviously.

Re: `NUMBER_OF_TOP_VALUES`: ✅

Reply:
Okay, I understand that all three are required, but I think there are two different things we want to customize:
1. Number of rows in the table (set in `size(limit)`)
2. Number of top values inside the visualization (which is used to highlight the _n_ top values in the table, and is also used to select which fields will be in the pie chart). Those settings are handled by `this.group.top(limit)` and `slicesCap(this.props.limit)`

Currently with only one setting for everything, and being 50 the default, the table will only display 50 values, all of them will be marked as "top values", and the pie chart will display 51 values (the 50 "top values" plus the others categories, if we don't limit the amount of data somewhere else). I think this may be a particular case of the settings, but using the visualization as it was before should still be possible and makes sense in my opinion.